added new two options namely "titleElement" and "additionalProperties

talatkuyuk opened the pull request:
Hi, @koka831.

I added new two options namely `titleElement` and `additionalProperties` in order to increase the customization as we have coordinated. 

I have written **additional tests** for the new options as well. 

I also exported the interface `CustomContainerOptions` in order to provide better type for the users who suffers type-matching for the new options, which I also needed.

koka831 commented on README.md:
````diff
@@ -58,23 +58,71 @@ import remark from "remark";
+</article>
+```
+
+**Note :** The `containerTag` is not prefered to be a `span` or similar, if there is an inner title `div` element. This may cause a problem because of a `div` element under a `span` element.
````
:+1:

koka831 commented on src/index.ts:
```diff
@@ -17,11 +17,21 @@ interface CustomContainerOptions {
 const DEFAULT_SETTINGS: CustomContainerOptions = {
   className: "remark-container",
   containerTag: "div",
+  titleElement: {},
```
[nit] `{ className: [] }` maybe better?

koka831 commented on src/index.ts:
```diff
@@ -17,11 +17,21 @@ interface CustomContainerOptions {
   className: "remark-container",
   containerTag: "div",
+  titleElement: {},
+  additionalProperties: undefined,
```
IMHO we can use `{}` for its default value for convenience.

talatkuyuk replied:
Do you mean the {} for the titleElement or the additionalProperties ?

koka831 replied:
I see it makes sense.

koka831 commented on src/index.ts:
```diff
@@ -3,12 +3,12 @@ import { visit } from "unist-util-visit";
 import type { Node, Literal, Parent } from "unist";
 import type { Paragraph } from "mdast";
-import { Data } from "vfile";
+import type { Data } from "vfile";
```
:+1:

koka831 commented on src/__tests__/options2.test.ts:
```diff
@@ -0,0 +1,54 @@
+import remark2rehype from "remark-rehype";
+import stringify from "rehype-stringify";
+
+import plugin, {type CustomContainerOptions} from "..";
```
[nit] 

```suggestion
import plugin, { type CustomContainerOptions } from "..";
```

koka831 commented on src/__tests__/options2.test.ts:
```diff
@@ -0,0 +1,54 @@
+
+const compiler: Processor = remark()
+  .use(gfm)
+  .use(plugin, {
+    additionalProperties: (className, title) => {}
+  } as CustomContainerOptions)
```
may I ask you to add a test without any options, just confirm `use(plugin)` works well without breaking change?

koka831 reviewed (commented):
Overall looks good, with some nits.
Thank you for the PR.

talatkuyuk commented:
Could you have a look at my comments on the conversations about **default value of the options**? Let's finalize the issue.

I fixed the lines you mentioned. I added the requested additional test.  After you agree with my comments I will make a second commit for the issue.

koka831 commented:
Hi, sorry for being late.
I'll review it in this weekend.
Thank you for patience:)

koka831 reviewed (approved):
LGTM, thanks.
I'll merge it after reviewing options4.test.

talatkuyuk commented:
Thank you for your positive review, I've pushed the additional commit.

koka831 commented:
Thank you @talatkuyuk.
 Could you try to relax these warnings? https://github.com/koka831/remark-custom-container/actions/runs/3664272373/jobs/6194702932

talatkuyuk commented:
I've pushed the third commit to fix the lint errors by making the parameters in the option "additionalProperties" **as optional**, which is more appropriate

koka831 reviewed (approved):
Looks good, I really appreciate your contribution.
I'll merge it and plan next release.
